### PR TITLE
git credential helper for git lfs auth

### PR DESCRIPTION
**Pull request by faza**

### CODE REVIEW

File: cmd/git-credential-gitopia/main.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	}
+
+	switch action {
+
+	case actionGet:
+		handleGetAction()
+
+	case actionErase, actionStore:
+		// Defined action but unsupported by this provider
+		return
```
Comment (ProgramCpp) [inline cmd/git-credential-gitopia/main.go:35]: why do we not return error here?

File: cmd/git-remote-gitopia/gitopia.go [hunk 2]
```diff
@@ -217,171 +122,79 @@ func (h *GitopiaHandler) List(remote *core.Remote, forPush bool) ([]string, erro
...
+	case wallet.ENV_VAR:
+		remote.Logger.Printf("Loaded Gitopia wallet file, path: %s, address: %s\n", os.Getenv("GITOPIA_WALLET"), h.wallet.Address())
+	case wallet.GITHUB_SEC:
+		remote.Logger.Printf("Loaded Gitopia wallet from GitHub secret, wallet address: %s\n", h.wallet.Address())
+	case wallet.LEDGER:
+		remote.Logger.Printf("Using Ledger device, wallet address: %s\n", h.wallet.Address())
+	case wallet.KEYRING_BACKEND:
+		keyName := h.wallet.(wallet.OSKeyring).KeyName()
+		remote.Logger.Printf("Using OS keyring, key name: %s, wallet address: %s\n", keyName, h.wallet.Address())
 	default:
```
Comment (ProgramCpp) [inline cmd/git-remote-gitopia/gitopia.go:193]: we can have an api in the wallet interface and just call `wallet.Info()` maybe?

File: cmd/git-remote-gitopia/gitopia.go [hunk 3]
```diff
@@ -434,114 +237,46 @@ func (h *GitopiaHandler) Push(remote *core.Remote, refsToPush []core.RefToPush)
...
 		if err != nil {
-			return nil, fmt.Errorf("fatal: error signing tx, %s", err.Error())
+			return nil, errors.Wrap(err, "error signing data")
 		}
 
-		txBz, err := encConf.TxConfig.TxJSONEncoder()(tx)
-		if err != nil {
-			return nil, fmt.Errorf("fatal: error encoding tx, %s", err.Error())
+		if h.wallet.Type() == wallet.LEDGER {
+			remote.Logger.Println("Please sign the git server request on your ledger device.")
```
Comment (ProgramCpp) [inline cmd/git-remote-gitopia/gitopia.go]: after signing on line 241, why do we prompt again?

File: cmd/git-remote-gitopia/gitopia.go [hunk 4]
```diff
@@ -582,100 +308,59 @@ func (h *GitopiaHandler) Push(remote *core.Remote, refsToPush []core.RefToPush)
...
+		msg = append(msg, gitopiaTypes.NewMsgMultiDeleteTag(h.wallet.Address(), gitopiaTypes.RepositoryId{
 			Id:   h.remoteRepository.Owner.Id,
 			Name: h.remoteRepository.Name,
 		}, deleteTags))
 	}
 
-	if h.secType == KEYRING_BACKEND {
-		account, err := h.kb.cc.Account(h.kb.key)
-		if err != nil {
+	switch v := h.wallet.(type) {
```
Comment (ProgramCpp) [inline cmd/git-remote-gitopia/gitopia.go]: keep `signAndBroadcast` signature same and simply call it? no need for switch again. we can initialize the grpcconn inside the type itself

**Review by ProgramCpp — APPROVED** [review]

File: cmd/git-remote-gitopia/gitopia.go (hunk 3 above)
Comment (faza) [reply cmd/git-remote-gitopia/gitopia.go]: move the prompt before the sign function

**Review by ProgramCpp — APPROVED** [review]

**Pull request merged**